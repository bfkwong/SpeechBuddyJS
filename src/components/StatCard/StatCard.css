.card-padding {
  margin: 15px 0px 5px 0px;
  border: none !important;
}
