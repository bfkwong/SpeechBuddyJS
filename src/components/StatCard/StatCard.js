import React from "react";
import { Card } from "react-bootstrap";
import "./StatCard.css";

function StatCard(props) {
  return (
    <Card className="card-padding">
      <Card.Body>
        <Card.Title data-testid="SC_value">{props.value}</Card.Title>
        <Card.Subtitle className="mb-2 text-muted" data-testid="SC_description">
          {props.description}
        </Card.Subtitle>
      </Card.Body>
    </Card>
  );
}

export default StatCard;
