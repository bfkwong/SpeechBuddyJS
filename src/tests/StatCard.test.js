//Bryan Kwong

import React from "react";
import { render } from "@testing-library/react";
import StatCard from "../components/StatCard/StatCard";
import renderer from "react-test-renderer";
import { unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

describe("StatCard Component", () => {
  let container;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    if (container.hasChildNodes()) {
      unmountComponentAtNode(container);
    }
    container.remove();
    container = null;
  });

  test("StatCard renders without failure", () => {
    render(<StatCard />, container);
  });

  test("StatCard snapshot test", () => {
    const appTree = renderer.create(<StatCard />).toJSON();
    expect(appTree).toMatchSnapshot();
  });

  test("StatCard snapshot test", () => {
    const appTree = renderer
      .create(<StatCard value={100} description={"Test"} />)
      .toJSON();
    expect(appTree).toMatchSnapshot();
  });

  test("StatCard value and description", () => {
    act(() => {
      render(<StatCard value={100} description={"Test"} />, container);
    });

    const SC_value = document.querySelector("[data-testid=SC_value]");
    expect(SC_value.innerHTML).toBe("100");

    const SC_description = document.querySelector(
      "[data-testid=SC_description]"
    );
    expect(SC_description.innerHTML).toBe("Test");
  });
});
