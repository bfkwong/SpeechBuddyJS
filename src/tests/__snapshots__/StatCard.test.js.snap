// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatCard Component StatCard snapshot test 1`] = `
<div
  className="card-padding card"
>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
      data-testid="SC_value"
    />
    <div
      className="mb-2 text-muted card-subtitle h6"
      data-testid="SC_description"
    />
  </div>
</div>
`;

exports[`StatCard Component StatCard snapshot test 2`] = `
<div
  className="card-padding card"
>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
      data-testid="SC_value"
    >
      100
    </div>
    <div
      className="mb-2 text-muted card-subtitle h6"
      data-testid="SC_description"
    >
      Test
    </div>
  </div>
</div>
`;
